src/pydicom/encaps.py

```diff
@@ -1,509 +1,624 @@
     """
-    if not fp.is_little_endian:
-        raise ValueError("'fp.is_little_endian' must be True")
-
-    tag = Tag(fp.read_tag())
+    if isinstance(buffer, bytes):
+        buffer = BytesIO(buffer)
 
-    if tag != 0xFFFEE000:
+    group, elem = unpack(f"{endianness}HH", buffer.read(4))
+    if group << 16 | elem != 0xFFFEE000:
         raise ValueError(
-            f"Unexpected tag '{tag}' when parsing the Basic Table Offset item"
+            f"Unexpected tag '{Tag(group, elem)}' when parsing the Basic Table Offset item"
         )
 
-    length = fp.read_UL()
+    length = unpack(f"{endianness}L", buffer.read(4))[0]
     if length % 4:
         raise ValueError(
             "The length of the Basic Offset Table item is not a multiple of 4"
         )
 
-    offsets = []
-    # Always return at least a 0 offset
     if length == 0:
-        offsets.append(0)
+        return []
 
-    for ii in range(length // 4):
-        offsets.append(fp.read_UL())
+    return list(unpack(f"{endianness}{length // 4}L", buffer.read(length)))
 
-    return bool(length), offsets
 
+def parse_fragments(
+    buffer: bytes | ReadableBuffer, /, *, endianness: str = "<"
+) -> tuple[int, list[int]]:
+    """Return the number of fragments and their positions in `buffer`.
 
-def get_nr_fragments(fp: DicomIO) -> int:
-    """Return the number of fragments in `fp`.
+    .. versionadded:: 3.0
+
+    Parameters
+    ----------
+    buffer : bytes | readable buffer
+        A buffer containing the encapsulated frame data, starting at the first
+        byte of item tag for a fragment, such as after the end of the Basic
+        Basic Offset Table. May be :class:`bytes` or an object with ``read()``,
+        ``tell()`` and ``seek()`` methods. If the latter then the offset will
+        be reset to the starting position afterwards.
+    endianness : str, optional
+        If ``"<"`` (default) then the encapsulated data uses little endian
+        encoding, otherwise if ``">"`` it uses big endian encoding.
 
-    .. versionadded:: 1.4
+    Returns
+    -------
+    tuple[int, list[int]]
+        The number of fragments and the absolute offset position of the first
+        byte of the item tag for each fragment in `buffer`.
     """
-    if not fp.is_little_endian:
-        raise ValueError("'fp.is_little_endian' must be True")
+    if isinstance(buffer, bytes):
+        buffer = BytesIO(buffer)
+
+    start_offset = buffer.tell()
 
     nr_fragments = 0
-    start = fp.tell()
+    fragment_offsets = []
     while True:
         try:
-            tag = Tag(fp.read_tag())
-        except EOFError:
+            group, elem = unpack(f"{endianness}HH", buffer.read(4))
+        except Exception:
             break
 
+        tag = group << 16 | elem
         if tag == 0xFFFEE000:
-            # Item
-            length = fp.read_UL()
+            length = unpack(f"{endianness}L", buffer.read(4))[0]
```

Comment:
If that fails because you are at the end, you will probably get a `struct.error`, which is not very nice for the user. Maybe handle this by raising a more readable exception?

Reply:
Done